server/src/main/java/org/kaazing/gateway/server/context/resolve/GatewayContextResolver.java

```diff
@@ -634,6 +634,13 @@ private DefaultServiceDefaultsContext resolveServiceDefaults(ServiceDefaultsType
             ConnectOptionsContext connectOptionsContext =
                     new DefaultConnectOptionsContext(connectOptions, defaultConnectOptions);
 
+            if (serviceType.equals("http.proxy") && acceptOptions == null && connectOptions == null) {
```

Comment:
Can we do this check in HttpProxyService#init(). Currently gateway server does many things which it shouldn't do, instead should done in specific components.